test/factories.rb

```diff
@@ -12,8 +12,8 @@
 		association :user
 		name "go go go"
 		description "it's fun"
-		start_location "123 Sesame Street, New York, NY, USA"
-		finish_location "456 Zoom Avenue, Los Angeles, CA, USA"
+		association :start_location, :factory => :address
+		association :finish_location, :factory => :address
```

Comment:
Your schema, unlike the nomster app, has the address as a separate database table.  Because of this you need to build up a whole new model to connect the association in your app, a string won't cut it.